services/server/test/integration/apiv1/verification-handlers/private.stateless.spec.ts

```diff
@@ -0,0 +1,574 @@
+      .to.equal("perfect");
+    chai
+      .expect(sourcifyMatchesResult.rows[0].creation_match)
+      .to.equal("perfect");
+
+    // Verify all other tables contain only the new contract data
+    // Check compiled_contracts - should only have the new compilation
+    const compiledContractsResult = await serverFixture.sourcifyDatabase.query(
+      "SELECT COUNT(*) as count, name FROM compiled_contracts GROUP BY name",
+    );
+    chai.expect(compiledContractsResult.rows[0].name).to.equal("Storage");
+    chai.expect(parseInt(compiledContractsResult.rows[0].count)).to.equal(1);
+
+    // Check compiled_contracts_sources - should only have sources for the new compilation
+    const compiledContractsSourcesResult =
+      await serverFixture.sourcifyDatabase.query(
+        "SELECT COUNT(*) as count FROM compiled_contracts_sources",
+      );
+    chai
+      .expect(parseInt(compiledContractsSourcesResult.rows[0].count))
+      .to.equal(1);
+
+    // Check sources - should only have sources for the new contract
+    const sourcesResult = await serverFixture.sourcifyDatabase.query(
+      "SELECT COUNT(*) as count FROM sources",
+    );
+    chai.expect(parseInt(sourcesResult.rows[0].count)).to.equal(1);
+
+    // Check contract_deployments - should only have the new deployment
+    const contractDeploymentsResult =
+      await serverFixture.sourcifyDatabase.query(
+        "SELECT COUNT(*) as count, encode(address, 'hex') as address FROM contract_deployments GROUP BY address",
+      );
+    chai
+      .expect(contractDeploymentsResult.rows[0].address.toLowerCase())
+      .to.equal(chainFixture.defaultContractAddress.toLowerCase().substring(2));
+    chai.expect(parseInt(contractDeploymentsResult.rows[0].count)).to.equal(1);
+
+    // Check contracts - should only have the new contract
+    const contractsResult = await serverFixture.sourcifyDatabase.query(
+      "SELECT COUNT(*) as count FROM contracts",
+    );
+    chai.expect(parseInt(contractsResult.rows[0].count)).to.equal(1);
+
+    // Check code - should only have code for the new contract (creation + runtime)
+    const codeResult = await serverFixture.sourcifyDatabase.query(
+      "SELECT COUNT(*) as count FROM code",
+    );
+    // Should have exactly 2 code entries: creation and runtime bytecode
+    chai.expect(parseInt(codeResult.rows[0].count)).to.equal(2);
+  });
+
+  it("should replace contract using existing database compilation (forceCompilation: false) and restore creation_match", async () => {
+    // First, verify with perfect match
+    const res = await chai
+      .request(serverFixture.server.app)
+      .post("/")
+      .field("address", chainFixture.defaultContractAddress)
+      .field("chain", chainFixture.chainId)
+      .field("creatorTxHash", chainFixture.defaultContractCreatorTx)
+      .attach("files", chainFixture.defaultContractMetadata, "metadata.json")
+      .attach("files", chainFixture.defaultContractSource);
```

Comment:
Why not use the `verifyContract` utility from helpers.ts here and in the other tests?